skyline-vscode/react-ui/src/sections/EnergyConsumption.js

```diff
@@ -0,0 +1,268 @@
+import React, { useState } from "react";
+import { Container, Row, Col, Spinner, Card, Form } from "react-bootstrap";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import {
+  faGlobeAmericas,
+  faCar,
+  faHome,
+  faMobileAlt,
+} from "@fortawesome/free-solid-svg-icons";
+import styled from "styled-components";
+import Subheader from "../Subheader";
+import PieGraph from "../components/PieGraph";
+import StackedBarGraph from "../components/StackedBarGraph";
+import { energy_data, unitScale } from "../utils";
+
+const EnergyConsumption = ({ energyData }) => {
+  const numIterations = 10000; 
+
+  const cpu_color = "#5499c7";
+  const cpu_color_opacity = "rgba(84,153,199,0.55)";
+  const gpu_color = "#17a589";
+  const gpu_color_opacity = "rgba(23,165,137,0.55)";
+
+  let total = null;
```

Comment:
What happens if there is no empty data from the response?

Reply:
all the variables are initiated as null, but depending on the data received they are updated one by one. At the end the component only shows information that is received from the backend